clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/config/BootstrapKubernetesConfigurationProvider.java

```diff
@@ -0,0 +1,148 @@
+import org.springframework.core.env.ConfigurableEnvironment;
+import org.springframework.core.env.PropertySource;
+
+/**
+ * For larger number of Kubernetes accounts, SpringBoot implementation of properties binding is
+ * inefficient, hence a custom constructor for KubernetesConfigurationProperties is written. This
+ * class fetches the flattened kubernetes properties from Spring Cloud Config's
+ * BootstrapPropertySource and creates a KubernetesConfigurationProperties object. Also adds support
+ * to existing "configserver:" feature along with caching ability. The main objective of this
+ * implementation is to reduce the clouddriver loading time when large number of dynamic accounts
+ * are configured. So some of the features provided by SpringBoot and clouddriver modules do not
+ * work. Here are the limitations: 1. The properties in the yaml file must use camel case 2.
+ * Property placeholders don't work. 3. "encryptedFile:" and "encrypted:" notations don't work 4.
+ * "configserver:" works only for kubeconfig files
+ */
+public class BootstrapKubernetesConfigurationProvider implements KubernetesConfigurationProvider {
+
+  private final ConfigurableApplicationContext applicationContext;
+  private CloudConfigResourceService configResourceService;
+  private Map<String, String> configServerCache;
+
+  public BootstrapKubernetesConfigurationProvider(
+      ConfigurableApplicationContext applicationContext) {
+    this.applicationContext = applicationContext;
+  }
+
+  @Override
+  public KubernetesConfigurationProperties getKubernetesConfigurationProperties() {
+    return getKubernetesConfigurationProperties(getPropertiesMap());
+  }
+
+  @SuppressWarnings("unchecked")
+  public KubernetesConfigurationProperties getKubernetesConfigurationProperties(
+      Map<String, Object> kubernetesPropertiesMap) {
+    ObjectMapper objectMapper = new ObjectMapper();
+    // remove the keys having blank string values
+    kubernetesPropertiesMap.values().removeAll(Collections.singleton(""));
+
+    Map<String, Object> propertiesMap =
+        (Map<String, Object>)
+            JsonUnflattener.unflattenAsMap(kubernetesPropertiesMap).get("kubernetes");
+
+    KubernetesConfigurationProperties kubernetesConfigurationProperties =
+        objectMapper.convertValue(propertiesMap, KubernetesConfigurationProperties.class);
+
+    kubernetesConfigurationProperties
+        .getAccounts()
+        .forEach(
+            acc -> {
+              if (acc.getKubeconfigFile().startsWith("configserver:")) {
+                acc.setKubeconfigFile(resolveConfigServerFilePath(acc.getKubeconfigFile()));
+              }
+            });
+    return kubernetesConfigurationProperties;
+  }
+
+  @SuppressWarnings("unchecked")
+  private Map<String, Object> getPropertiesMap() {
+    ConfigurableEnvironment environment = applicationContext.getEnvironment();
+    Map<String, Object> map;
+
+    for (PropertySource<?> propertySource : environment.getPropertySources()) {
```

Comment:
If the environment has multiple property sources, it looks like we will be returning immediately on the first here.  What happens in the case that there is a `spinnaker.yml`, a `clouddriver.yml`, and `clouddriver-local.yml`.  It seems like this logic will not be properly merging these (and depending on the order will just pick one).

Reply:
This is the same behaviour as existing clouddriver. If there are multiple files having kubernetes accounts, only the one having highest priority wins. The accounts don't get merged from different files.

Reply:
Sounds good as long as this has been tested for cases where the kubernetes accounts live in any of the above files.  Definitely I realize currently there's no merging behavior, but currently I could put my accounts in any of `spinnaker.yml`, `clouddriver.yml`, or `clouddriver-local.yml` as long as I only pick one, and it will work.  What I was unsure of is whether after these changes only the highest-priority source would be used.  For example, if `clouddriver-local.yml` is highest priority and does not define kubernetes accounts, will we still use the ones from `clouddriver.yml`?